core/src/test/scala/unit/kafka/server/OffsetCommitTest.scala

```diff
@@ -284,10 +313,41 @@ class OffsetCommitTest extends ZooKeeperTestHarness {
     Thread.sleep(retentionCheckInterval * 2)
     assertEquals(-1L, simpleConsumer.fetchOffsets(fetchRequest).requestInfo.get(topicPartition).get.offset)
 
+    ShutdownServerAndConsumer()
+  }
+
+  @Test
+  def testOffsetExpirationInOldBrokerVersion() {
+    StartServerAndConsumer(KAFKA_1_1_IV0)
+
+    // set up topic partition
+    val topic = "topic"
+    val topicPartition = TopicAndPartition(topic, 0)
+    createTopic(zkClient, topic, servers = Seq(server), numPartitions = 1)
+
```

Comment:
Can you add a comment on what this test is trying to test?

Reply:
Sure. Done.